tests/test_view/test_views.py

```diff
@@ -350,39 +163,56 @@ async def check_ws_response2(ws):
     assert data == {"status": "success", "balance": "0.01"}
 
 
-def test_ping_email(client: TestClient, token: str):
-    resp0 = client.get("/stores/55/ping")
-    assert resp0.status_code == 401
-    resp = client.get("/stores/55/ping", headers={"Authorization": f"Bearer {token}"})
-    assert resp.status_code == 404
-    resp1 = client.get("/stores/2/ping", headers={"Authorization": f"Bearer {token}"})
-    assert resp1.status_code == 200
-    assert not resp1.json()
+@Parametrization.autodetect_parameters()
+@Parametrization.case(name="non-exist-store-unauthorized", store_exists=True, authorized=False)
+@Parametrization.case(name="non-exist-store-authorized", store_exists=False, authorized=True)
+@Parametrization.case(name="store-unauthorized", store_exists=True, authorized=False)
+@Parametrization.case(name="store-authorized", store_exists=True, authorized=True)
+def test_ping_email(client: TestClient, token: str, store, store_exists, authorized):
+    store_id = store.id if store_exists else 999
+    resp = client.get(f"/stores/{store_id}/ping", headers={"Authorization": f"Bearer {token}"} if authorized else {})
+    if authorized:
+        if store_exists:
+            assert resp.status_code == 200
+            assert not resp.json()
+        else:
+            assert resp.status_code == 404
+    else:
+        assert resp.status_code == 401
 
 
-def test_user_stats(client: TestClient, token: str):
-    resp = client.get("/users/stats", headers={"Authorization": f"Bearer {token}"})
+@pytest.mark.asyncio
+async def test_user_stats(async_client, user: dict, token, store_wallet: dict):
+    store_id = store_wallet["store"].id
+    await create_product(user["id"], store_id=store_id)
+    await create_invoice(user["id"], store_id=store_id)
+    await create_invoice(user["id"], store_id=store_id)
+    resp = await async_client.get("/users/stats", headers={"Authorization": f"Bearer {token}"})
     assert resp.status_code == 200
     assert resp.json() == {
-        "wallets": 3,
-        "stores": 2,
-        "discounts": 1,
+        "discounts": 0,
+        "invoices": 2,
+        "notifications": 0,
         "products": 1,
-        "invoices": 0,
-        "notifications": 1,
-        "templates": 1,
+        "stores": 1,
+        "templates": 0,
+        "wallets": 1,
         "balance": 0.01,
     }
 
 
-def test_categories(client: TestClient):
-    assert client.get("/products/categories").status_code == 422
-    resp = client.get("/products/categories?store=1")
-    resp2 = client.get("/products/categories?store=2")
+@pytest.mark.asyncio
+@Parametrization.autodetect_parameters()
+@Parametrization.case(name="single", categories=["all"])
```

Comment:
Note: as far as I see, this test would create a product with category `all`. The intention of first test was to ensure that `all` category is returned in any case, even when no categories are available